### PR TITLE
added topic interpretability to eval metrics

### PR DESCRIPTION
**Pull request by Adityakolluru7**

### CODE REVIEW

File: octis/evaluation_metrics/interpretability_metrics.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        super().__init__()
+        if texts is None:
+            raise Exception("There are no texts in the document")
+        else:
+            self._texts = texts
+        self._dictionary = Dictionary(self._texts)
+        self.topk = topk
+        self.coherence_measure = coherence_measure
+        c_npmi = Coherence(texts, topk=topk, measure=coherence_measure)
+        topic_diversity = TopicDiversity(topk=topk)
```
Comment (stepgazaille) [inline octis/evaluation_metrics/interpretability_metrics.py:32]: Lines 32 and 34 could be merged into one instruction

File: octis/evaluation_metrics/interpretability_metrics.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from octis.evaluation_metrics.diversity_metrics import TopicDiversity
+from octis.evaluation_metrics.coherence_metrics import Coherence
+
+
+class TopicInterpretability(AbstractMetric):
+    def __init__(
+        self,  # noqa
+        texts: str = None,
+        topk: int = 10,
+        coherence_measure: str = "c_npmi",
```
Comment (stepgazaille) [inline octis/evaluation_metrics/interpretability_metrics.py:12]: In our last conversation we came to the conclusion that our TI would be (NPMI + 1) * TD, so we shouldn't allow to  to use another `coherence_measure`

File: octis/evaluation_metrics/interpretability_metrics.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        topk: int = 10
+    ) -> None:
+        """
+        Initialize metric
+        Parameters
+        ----------
+        texts : list of documents (list of lists of strings)
+        topk : how many most likely words to consider in
+        the evaluation
+        measure : (default 'c_npmi') measure to use.
```
Comment (gelimcgill) [inline octis/evaluation_metrics/interpretability_metrics.py:20]: this is not an input argument. We can just remote this line.

**Review by gelimcgill — CHANGES_REQUESTED** [review]

File: octis/evaluation_metrics/interpretability_metrics.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        """
+        super().__init__()
+        if texts is None:
+            raise Exception("There are no texts in the document")
+        else:
+            self._texts = texts
+        self._dictionary = Dictionary(self._texts)
+        self.topk = topk
```
[the remark this answers is not in the document]
Comment (stepgazaille) [reply octis/evaluation_metrics/interpretability_metrics.py:28]: `self._dictionary` and `self.topk` are not directly used within the `TopicInterpretability` class.
Does OCTIS need them somewhere else?

File: octis/evaluation_metrics/interpretability_metrics.py (hunk 3 above)
Comment (Adityakolluru7) [reply octis/evaluation_metrics/interpretability_metrics.py:20]: Yes, sure!

File: octis/evaluation_metrics/interpretability_metrics.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Adityakolluru7) [reply octis/evaluation_metrics/interpretability_metrics.py:28]: `self.topk` is used in lines 29 and 30 for both Coherence and Topic Diversity, I think I can remove `self._dictionary`

[the remark this answers is not in the document]
Comment (stepgazaille) [reply octis/evaluation_metrics/interpretability_metrics.py:28]: My point is they are used right here in the constructor and never again, so no need to store them in an attribute

**Review by stepgazaille — APPROVED** [review]

**Pull request merged**